### ISSUE TITLE
resolve services by alias, not ip

### ISSUE THREAD

**Issue by ifokeev** [mesudip/nginx-proxy#18]
I'm wondering why nginx-proxy is resolving proxy_pass as IP but not service alias name? It looks like I have issues with nginx-proxy on multi node swarm setup

**Comment by mesudip** [comment]
Yeh, that's a known issue. Nginx-proxy uses ip instead of dns, and it can see containers running on the same host only.

Advantage of this is, nginx doesn't have to resolve DNS and the requests don't hop to another other server.

In my swarm setup I am using a workaround. I run  running nginx-proxy with `mode=global` and restrict the placement of the containers to be exposed and then I update my dns for that service to the node on which It will run.

**Comment by ifokeev** [comment]
@mesudip but it requires a simple fix to work with aliases

**Comment by mesudip** [comment]
Indeed, may be I should make it possible to configure what to use with an Environment Variable

**Comment by ifokeev** [comment]
@mesudip it would be wonderful if you may add the fix
